frontend/src/components/onboarding/FailUserModal.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from "react";
+import { useMutation } from "@apollo/client";
+import { useHistory } from "react-router-dom";
+import {
+  Button,
+  Flex,
+  Grid,
+  GridItem,
+  Heading,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  Textarea,
+} from "@chakra-ui/react";
+import {
+  FINISH_GRADING_STORY_TRANSLATION,
+  FinishGradingStoryTranslationResponse,
+} from "../../APIClients/mutations/StoryMutations";
+import {
+  GRADING_PAGE_FAIL_USER_CONFIRMATION,
+  GRADING_PAGE_FAIL_USER_BUTTON,
+} from "../../utils/Copy";
+import ConfirmationModal from "../utils/ConfirmationModal";
+
+export type FailUserModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  storyTranslationId: number;
+};
+
+const FailUserModal = ({
+  isOpen,
+  onClose,
+  storyTranslationId,
+}: FailUserModalProps) => {
+  const [enterFeedback, setEnterFeedback] = useState<boolean>(false);
+  const [testFeedback, setTestFeedback] = useState<string>("");
+
+  const history = useHistory();
+
+  const [finishGradingStoryTranslation] = useMutation<{
+    finishGradingStoryTranslation: FinishGradingStoryTranslationResponse;
+  }>(FINISH_GRADING_STORY_TRANSLATION);
+
+  const failGrade = async (feedback: string) => {
+    const result = await finishGradingStoryTranslation({
+      variables: {
+        storyTranslationTestId: storyTranslationId,
+        testFeedback: feedback,
+        testResult: "{}",
+      },
+    });
+    if (result.data?.finishGradingStoryTranslation.ok) {
+      history.push("/");
+    } else {
+      window.alert("Could not mark story translation as failed.");
+    }
+  };
+
+  return (
+    <>
+      <ConfirmationModal
+        confirmation={isOpen && !enterFeedback}
+        onConfirmationClick={() => setEnterFeedback(true)}
+        onClose={onClose}
+        confirmationMessage={GRADING_PAGE_FAIL_USER_CONFIRMATION}
+        buttonMessage={GRADING_PAGE_FAIL_USER_BUTTON}
+      />
+      <Modal
```

Comment:
this modal is the same as the modal from AssignTestGradeModal, mind extracting the component to another file and then using the component in here + in AssignTestGradeModal? main difference between the two modals is that back button doesnt exist on the fail flow